app/models/progress_event.rb

```diff
@@ -0,0 +1,19 @@
+class ProgressEvent < ActiveRecord::Base
+
+  # relationships
+  belongs_to :user, inverse_of: :progress_events
+  belongs_to :dataset_item, inverse_of: :progress_events
+
+  # add deleted_at and deleter_id
+  # TODO: ask anthony
+
+  # association validations
+  validates :user, existence: true
+  validates :dataset_item, existence: true
+
+  # field validations
+  validates :activity, inclusion: { in: ['viewed', 'played', 'annotated'] }
```

Comment:
add a comment here that the activity types allowed here are largely arbitrary and we expect to remove this restriction